docs/deepnote-format.md

```diff
@@ -0,0 +1,361 @@
+  - Chart blocks with visual query builder
+  - Input blocks for interactive parameters
+  - Big number blocks for KPIs
+  - Rich text blocks with advanced formatting
+
+### Execution modes
+
+**Jupyter:**
+
+- Sequential execution only
+- Cells executed in order or individually
+
+**Deepnote:**
+
+- **Block mode**: Execute blocks independently
+- **Downstream mode**: Automatically execute dependent blocks
+- Dependency tracking between blocks
+
+### Metadata and settings
+
+**Jupyter:**
+
+```json
+{
+  "metadata": {
+    "kernelspec": {
+      "name": "python3",
+      "display_name": "Python 3"
+    }
+  }
+}
+```
+
+**Deepnote:**
+
+```yaml
+project:
+  settings:
+    environment:
+      pythonVersion: "3.11"
+      customImage: "custom-image:latest"
+    requirements:
+      - "pandas>=2.0.0"
+    sqlCacheMaxAge: 3600
+```
+
+### Output storage
+
+**Jupyter:**
+
+- Outputs embedded in notebook JSON
+- Can make files very large
+- Binary data base64-encoded
+
+**Deepnote:**
+
+- Outputs stored in YAML format
+- Cleaner representation
+- Better for version control
+- Optional output persistence
+
+## Converting between formats
+
+### Jupyter to Deepnote
+
+Use the `@deepnote/convert` package to convert Jupyter notebooks:
+
+```bash
+# Convert a single notebook
+deepnote-convert notebook.ipynb
+
+# Convert multiple notebooks into one project
+deepnote-convert ./notebooks/ --projectName "My Project"
+
+# Specify output location
+deepnote-convert notebook.ipynb -o output.deepnote
+```
+
+**What gets converted:**
+
+- ✅ Code cells → Code blocks
+- ✅ Markdown cells → Markdown blocks
+- ✅ Cell outputs → Block outputs
+- ✅ Execution counts → Execution counts
+- ✅ Cell metadata → Block metadata
+
+**What doesn't convert:**
+
+- ⚠️ Jupyter widgets (may need manual adjustment)
+- ⚠️ Custom cell magics (may need adaptation)
+- ⚠️ Kernel-specific features
+
+### Deepnote to Jupyter
+
+Export from Deepnote to `.ipynb` format:
+
+```bash
+# Using Deepnote UI: Notebook menu → Download as .ipynb
+```
+
+**Conversion notes:**
+
+- SQL blocks → Converted to code cells with string representation
+- Chart blocks → Converted to markdown with chart description
+- Input blocks → Converted to code cells with default values
+- Rich text blocks → Converted to standard markdown
+
+## Example of a complete Deepnote file
+
+Here's a minimal but complete example:
+
+```yaml
+metadata:
+  createdAt: "2025-01-27T12:00:00Z"
+
+version: "1.0.0"
+
+project:
+  id: "abc-123-def-456"
+  name: "Sales Analysis"
+
+  notebooks:
+    - id: "notebook-001"
+      name: "Q4 Analysis"
+      executionMode: "block"
+      isModule: false
+      blocks:
+        - id: "block-001"
+          type: "markdown"
+          sortingKey: "1"
+          content: |
+            # Q4 Sales Analysis
+
+            This notebook analyzes sales data for Q4 2024.
+          version: 1
+          metadata: {}
+
+        - id: "block-002"
+          type: "code"
+          sortingKey: "2"
```

Comment:
Can we remove all of this (and further below) and link to converting-notebooks.md instead? We are duplicating content and it can become a nightmare to maintain.

Reply:
My bad 👍